### PR TITLE
SONAR-7681 Use public implementation of List in MyBatis

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/test/java/org/sonar/server/ce/ws/ActivityActionTest.java [hunk 1]
```diff
@@ -304,14 +322,18 @@ public void fail_if_date_is_not_well_formatted() {
...
-    userSession.setGlobalPermissions(GlobalPermissions.SYSTEM_ADMIN);
+    globalAdmin();
     TestResponse wsResponse = ws.newRequest()
       .setMediaType(MediaTypes.JSON)
       .execute();
 
     JsonAssert.assertJson(wsResponse.getInput()).isSimilarTo("{\"tasks\":[]}");
   }
 
+  private void globalAdmin() {
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/ce/ws/ActivityActionTest.java]: A UT is missing when a user have only project admin permission

File: sonar-db/src/main/java/org/sonar/db/component/ComponentTreeQuery.java [hunk 2]
```diff
@@ -39,7 +41,7 @@
   @CheckForNull
   private final String nameOrKeyQuery;
   @CheckForNull
-  private final Collection<String> qualifiers;
+  private final ArrayList<String> qualifiers;
```
Comment (simonbrandhof) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentTreeQuery.java]: should be `List`

**Comment by simonbrandhof** [comment]
the second commit LGTM

**Comment by julienlancelot** [comment]
For the permission commit, add UT for a project admin user and LGTM.

**Pull request merged**